daemon/anonymous-statistics.sh

```diff
@@ -0,0 +1,157 @@
+fi
+
+# -------------------------------------------------------------------------------------------------
+# detect the kernel
+
+KERNEL_NAME="$(uname -s)"
+KERNEL_VERSION="$(uname -r)"
+ARCHITECTURE="$(uname -m)"
+
+# -------------------------------------------------------------------------------------------------
+# detect the virtualization
+
+VIRTUALIZATION="unknown"
+VIRT_DETECTION="none"
+
+if [ ! -z "$(command -v systemd-detect-virt 2>/dev/null)" ]
+then
+    VIRTUALIZATION="$(systemd-detect-virt)"
+    VIRT_DETECTION="systemd-detect-virt"
+else
+    # /proc/1/sched exposes the host's pid of our init !
+    # http://stackoverflow.com/a/37016302
+    pid=$( head -n 1 </proc/1/sched 2>/dev/null | { IFS='(),#:' read -r _unused pid _unused _unused; echo $pid; } )
+    if [ ! -z "${pid}" ]; then
+        pid=$(( pid + 0 ))
+        [ ${pid} -ne 1 ] &&
+            VIRTUALIZATION="container" &&
+            VIRT_DETECTION="pid"
+    fi
+
+    # ubuntu and debian supply /bin/running-in-container
+    # https://www.apt-browse.org/browse/ubuntu/trusty/main/i386/upstart/1.12.1-0ubuntu4/file/bin/running-in-container
+    [ -x "/bin/running-in-container" ] &&
+        "/bin/running-in-container" >/dev/null 2>&1 &&
+        VIRTUALIZATION="container" &&
+        VIRT_DETECTION="/bin/running-in-container"
+
+    # lxc sets environment variable 'container'
+    [ ! -z "${container}" ] &&
+        VIRTUALIZATION="lxc" &&
+        VIRT_DETECTION="containerenv"
+
+    # docker creates /.dockerenv
+    # http://stackoverflow.com/a/25518345
+    [ -f "/.dockerenv" ] &&
+        VIRTUALIZATION="docker" &&
+        VIRT_DETECTION="dockerenv"
+fi
+
+# -------------------------------------------------------------------------------------------------
+# check netdata version
+
+if [ -z "${NETDATA_VERSION}" ]; then
+    NETDATA_VERSION="$(netdata -V 2>&1 | cut -d ' ' -f 2)"
+fi
+
+# -------------------------------------------------------------------------------------------------
+# check netdata unique id
+
+if [ -z "${NETDATA_REGISTRY_UNIQUE_ID}" ]; then
+    for dir in "$(netdata -W get global 'lib directory' '/var/lib/netdata' 2>/dev/null)" "/var/lib/netdata" "/opt/netdata/var/lib/netdata"; do
+        if [ -f "${dir}/registry/netdata.public.unique.id" ]; then
+            NETDATA_REGISTRY_UNIQUE_ID="$(cat "${dir}/registry/netdata.public.unique.id")"
+            break
+        fi
+    done
+fi
+
+# -------------------------------------------------------------------------------------------------
+# send the anonymous statistics to GA
+ACTION_DATA=$(echo "${ACTION_DATA}" | tr '"' "'")
+
+payload="&v=1&an=netdata&aip=1&t=pageview&_s=1&ds=statistics&tid=UA-64295674-3&cid=${NETDATA_REGISTRY_UNIQUE_ID}&t=event&av=${NETDATA_VERSION}&ec=${ACTION}&ea=${ACTION_RESULT}&dh=netdata-agent.my-netdata.io&dp=%2Fstatistics"
+
+curl -X POST -Ss --max-time 2 \
+--data-raw "${payload}"   					\
```

Comment:
nit: those whitespaces look unnecessary

Reply:
I was saying that in regard to whitespaces between parameter and `\`, not regarding whitespaces at the start of a line. Sorry for lack of clearance.